### ISSUE TITLE
Viewer Credentials authentication

### ISSUE THREAD

**Issue by barryhammen** [tableau/connector-plugin-sdk#488]
Name: Barry Hammen
Company: Immuta

I'm trying to use https://github.com/tableau/connector-plugin-sdk/pull/417 as an example to add support for Viewer Credentials to our connector. When I go to publish a workbook to a Tableau Server, the only options for authentication are "Prompt user" or "Embedded password". 

I've added `CAP_AUTH_KERBEROS_IMPERSONATE` to the manifest.
I'm using `<option name="Integrated" value="auth-integrated" />` for the `ComboBoxIntegreted` authentication mode in the connection dialog.

I'm able to successfully create the Data Source and build the workbook when picking "Integrated" as the option. And I can publish other built-in connectors (like PostgreSQL) to my Tableau Server instance using "Viewer Credentials".

What am I missing?
Thanks

**Comment by devanshsoni9** [comment]
Hi @barryhammen , 
Can you please make sure you have configured your Tableau Server as per the kerberos delegation articles 

1. For configuring Kerberos delegation on Tableau Server Linux, refer to this [article](https://help.tableau.com/current/server-linux/en-us/kerberos_delegation.htm). 
2. For configuring Kerberos delegation on Tableau Server Windows. in addition to the steps mentioned [here](https://help.tableau.com/current/server/en-us/kerberos_delegation.htm), also execute the below steps before running `tsm pending-changes apply`
```
    tsm configuration set -k native_api.datasource_impersonation_runas_principal -v <delegation/Run As account user or principal>
    tsm configuration set -k native_api.datasource_impersonation_runas_keytab_path -v <path-to-file>kerberos.keytab
```

Thanks 
Devansh

**Comment by devanshsoni9** [comment]
also, not to forget that #417 is not yet merged into release branch and we do not have officially  support kerberos for plugins ([supported-auth-modes](https://tableau.github.io/connector-plugin-sdk/docs/auth-modes)). As we had seen in the postgres use case, supporting kerberos might require passing driver specific properties. We encourage plugin developers to try out and we can make more instructions available as we come across use cases. 

For now, the starting point should be following the steps I mentioned in the previous comment.

**Comment by barryhammen** [comment]
Thanks. I was hoping maybe the supported auths section was just out of date :)

Tableau Server is configured for kerberos/viewer credentials. I can publish a dashboard using a PostgreSQL data source just fine.

**Comment by devanshsoni9** [comment]
alright, let me know what driver you are trying to use and what tableau desktop version you are using. I can try and reproduce the issue and see how we can help you out.

**Comment by barryhammen** [comment]
Using the Postgres ODBC driver with Tableau Desktop 2020.1.

**Comment by devanshsoni9** [comment]
The sample mentions support for kerberos using JDBC and that would require installation of   Tableau Desktop/Server 2020.2 which hasn't been released yet. 
I will test for ODBC in 2020.1 and see it if works.

**Comment by sidwray** [comment]
tfs 1197768

**Comment by jkoskela** [comment]
This will be fixed in Tableau 2021.3. If it is needed for an earlier release it can be backported by request.
